File: app/src/main/java/be/vergauwen/simon/androidqualitycontrol/core/di/ApplicationComponent.java
package be.vergauwen.simon.androidqualitycontrol.core.di;

import android.content.Context;
import be.vergauwen.simon.androidqualitycontrol.core.rx.Transformers;
import be.vergauwen.simon.androidqualitycontrol.core.service.GithubAPI;
import dagger.Component;

@ApplicationScope
@Component(modules = {ApplicationModule.class,ServiceModule.class})
public interface ApplicationComponent {
    Context getApplicationContext();
    Transformers getTransfomers();
    GithubAPI getGithubAPI();
}

File: app/src/main/kotlin/be/vergauwen/simon/androidqualitycontrol/core/di/ApplicationComponent.kt
package be.vergauwen.simon.androidqualitycontrol.core.di

import android.content.Context
import be.vergauwen.simon.androidqualitycontrol.core.rx.Transformers
import be.vergauwen.simon.androidqualitycontrol.core.service.GithubAPI
import dagger.Component

@ApplicationScope
@Component(modules = arrayOf(ApplicationModule::class,ServiceModule::class))
interface ApplicationComponent {
  val applicationContext: Context
  val transformers : Transformers
  val githubAPI : GithubAPI
}